proxy/envoy/discovery.go

```diff
@@ -396,41 +396,45 @@ func (ds *DiscoveryService) ListEndpoints(request *restful.Request, response *re
 		}
 		var err error
 		if out, err = json.MarshalIndent(hosts{Hosts: hostArray}, " ", " "); err != nil {
-			errorResponse(response, http.StatusInternalServerError, err.Error())
+			errorResponse(response, http.StatusInternalServerError, "EDS "+err.Error())
 			return
 		}
 		ds.sdsCache.updateCachedDiscoveryResponse(key, out)
 	}
 	writeResponse(response, out)
 }
 
-func (ds *DiscoveryService) parseRole(request *restful.Request) (proxy.Node, error) {
-	if sc := request.PathParameter(ServiceCluster); sc != ds.Mesh.IstioServiceCluster {
-		return proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, sc)
+func (ds *DiscoveryService) parseDiscoveryRequest(request *restful.Request) (string, string, proxy.Node, error) {
+	cluster := request.PathParameter(ServiceCluster)
+	// request has to match the IstioServiceCluster (default is "istio-proxy")
+	if cluster != ds.Mesh.IstioServiceCluster {
+		return cluster, "", proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, cluster)
```

Comment:
while you are at this, can you get rid of this check completely? Why do we need to check if the service cluster is istio-proxy? In VM case, this is most likely going to fail, if the user customizes their envoy startup. (cc @costinm )
This is an artificial requirement we imposed on ourselves that everything has to be called istio-proxy.